asserts/repair.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+)
+
+// Repair holds an repair assertion which allows running repair
+// code to fixup broken systems. It can be limited by series and models.
+type Repair struct {
+	assertionBase
+
+	series []string
+	models []string
+}
+
+// RepairID returns the "id" of the repair. It should be a short string
+// that follows a convention like "REPAIR-123". Similar to a CVE there
+// should be a public place to look up details about the repair-id
+// (e.g. the snapcraft forum).
+func (em *Repair) RepairID() string {
```

Comment:
To Gustavo's point I think that we may see two classes of repair assertions. Those that are issued by a particular brand to address an isolated issue (where isolated is broad enough to include all devices from that brand). The other kind may be issued by Canonical to fix all core systems (or at least check if all core systems require any kind of fix). I'd say that the order should be arranged such that all canonical assertions are applied before any other assertions are considered.

What do you think?